logic/and.js

```diff
@@ -0,0 +1,26 @@
+/** @license ISC License (c) copyright 2017 original and current authors */
+/** @author Ian Hofmann-Hicks (evil) */
+
+const isFunction = require('../predicates/isFunction')
+const isSameType = require('../predicates/isSameType')
+
+const Pred = require('../crocks/Pred')
+
+const curry = require('../helpers/curry')
+const identity = require('../combinators/identity')
+
+const ifElse = require('./ifElse')
+
+const predOrFunc =
+  ifElse(isFunction, identity, x => x.runWith)
+
+// and : (a -> Boolean) | Pred -> (a -> Boolean) | Pred -> a -> Boolean
+function and(f, g) {
+  if(!((isFunction(f) || isSameType(Pred, f)) && (isFunction(g) || isSameType(Pred, g)))) {
+    throw new TypeError('and: Preds or predicate functions required for first two arguments')
```

Comment:
I would like to use both.
a `Pred` is an actual `ADT`, a predicate is a function from `(a -> Boolean)` are you suggesting that there be two function, one for working with only predicate functions, and another with `Pred`?

In case you are wondering, here is the linky to [`Pred`](https://github.com/evilsoft/crocks/blob/master/crocks/Pred.js)